### ISSUE TITLE
Multi-module project with both android & kotlin modules

### ISSUE THREAD

**Issue by phcannesson** [Kotlin/kotlinx-kover#422]
I have a multi-module android app that uses both android & pure kotlin modules.

To simplify, let's say that we have three modules as follows : 

```mermaid
  graph TD;
      app-->data;
      app-->domain;
```

where :
- `app` and `data` are android modules, so we would use `koverHtmlReportRelease` to generate a report for the release build variant.
- `domain` is pure kotlin modules, so we would use `koverHtmlReport` to generate a report.

Coming from 0.6, my goal is to retrieve more or less the same coverage I had with the previous merged report.

➡️ If I try to generate reports individually, I get a lower coverage because tests from another modules are not taken into account in the coverage calculation : for example a test in `app` will not cover `domain` code.

➡️ I tried using the new `dependencies {}` DSL, in `app`, like this : 
```
koverReport {
    androidReports("prodRelease") {
        dependencies {
            kover(project(":domain"))
            kover(project(":data"))
        }
        ...
    }
}
```

But, in this case, the task `koverHtmlReport` only generates coverage for `domain`, and `koverHtmlReportRelease` only generates coverage for `app` and `data`.

What is the correct configuration I need to use ?

**Comment by shanshin** [comment]
Hi,
It is better to specify a block with dependencies in the root of the script
```
dependencies {
    kover(project(":domain"))
    kover(project(":data"))
}

koverReport {
    // configuring reports if necessary
}
```


To use the coverage of some Android build variant when running task `koverHtmlReport`, you should configure the default tasks:
```
koverReport {
    defaults {
        mergeWith("prodRelease") // or the name of any build variant needed
    }
}
```
[See also](https://kotlin.github.io/kotlinx-kover/gradle-plugin/#kotlin-multiplatform-project)

**Comment by phcannesson** [comment]
Thanks a lot for you answer.

I managed to make it work by doing the following : 
- applying the kover plugin on each module,
- specifying all dependencies in the app module kover configuration,
- launching the coverage task on the app module.

I hope this helps others !

**Comment by JCarlosR** [comment]
> I managed to make it work by doing the following :
> 
> * applying the kover plugin on each module,
> * specifying all dependencies in the app module kover configuration,
> * launching the coverage task on the app module.

Do you mind sharing an example of the coverage task please?

**Comment by stackunderflows** [comment]
> Thanks a lot for you answer.
> 
> I managed to make it work by doing the following :
> 
> * applying the kover plugin on each module,
> * specifying all dependencies in the app module kover configuration,
> * launching the coverage task on the app module.
> 
> I hope this helps others !

Yes! Worked for me too!

**Comment by Wirling** [comment]
I can't get this to work.
For example I have `:data` which is an android module, and `:domain` which is a kotlin module.
- I applied the kover plugin to all modules.
- I specified a dependency `kover(project(":domain"))` in the `:data` module
If I launch `:data:koverHtmlReportDebug` it doesn't run the coverage in `:domain`
If I launch `:data:koverHtmlReport` it does run the coverage in `:domain` but is also takes longer because this job runs for each variant.
So to me it seems when I run `:data:koverHtmlReportDebug` it doesn't trigger `:domain` because it can't find the debug variant.

I also tried creating a custom variant so I can specify the correct variant manually, but since domain has no variant, this doesn't work either.
